# Determination - Deterministic rendering environment for white-axe's music
# Copyright (C) 2024 Liu Hao <whiteaxe@tuta.io>
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.

# This patch forces all of Carla's libraries to be built as static libraries.

--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -61,11 +61,11 @@ if(${CARLA_BUILD_FRAMEWORKS} AND NOT APPLE)
   mark_as_advanced(CARLA_BUILD_FRAMEWORKS)
 endif()
 
-if(${CARLA_BUILD_STATIC})
+#if(${CARLA_BUILD_STATIC})
   set(CARLA_LIB_TYPE STATIC)
-else()
-  set(CARLA_LIB_TYPE SHARED)
-endif()
+#else()
+#  set(CARLA_LIB_TYPE SHARED)
+#endif()
 
 if(${CARLA_BUILD_FRAMEWORKS})
   execute_process(COMMAND date -u +%y%m%d%H%M%S OUTPUT_VARIABLE CARLA_DATE OUTPUT_STRIP_TRAILING_WHITESPACE)
